src/fetch_business_partners_info.py

```diff
@@ -0,0 +1,104 @@
+import os
+import json
+import datetime
+import time
+import pandas as pd
+from unicodedata import normalize
+from selenium import webdriver
+from bs4 import BeautifulSoup
+
+
+def selenium_driver_path():
+    for path in ['chromedriver.exe', 'chromedriver']:
+        if os.path.exists(path):
+            return path
+    raise BaseException('ChromeDriver not found')
+
+
+def normalize_name(name):
+    return normalize('NFKD', name).encode('ASCII', 'ignore').decode('utf-8').lower()
+
+
+def make_google_cache_url(name):
+    return 'http://webcache.googleusercontent.com/search?btnG=Search&q=cache:www.consultasocio.com/q/sa/%s' % \
+           normalize_name(name).replace(' ', '-')
+
+
+def make_consulta_socio_url(name):
+    return 'http://www.consultasocio.com/q/sa/%s' % normalize_name(name).replace(' ', '-')
+
+
+def extract_info_from_page(url, info, driver, error_msg):
+    # Open the page using Selenium in order to collect the full HTML to be parsed by BeautifulSoup
+    driver.get(url)
+    driver.switch_to.default_content()
+    soup = BeautifulSoup(driver.page_source, 'html.parser')
+
+    # Somethimes google can ask for captcha, so you should type it!
+    if 'To continue, please type the characters below:' in soup.text:
+        while 'To continue, please type the characters below:' in soup.text:
+            driver.switch_to.default_content()
+            soup = BeautifulSoup(driver.page_source, 'html.parser')
+            time.sleep(1)
```

Comment:
Sorry about this last comment, I've talked to Irio but forgot to edit it before sending the review. Anyway: even if it's not the ideal it's not a big issue to depende on human interaction by now. Just a few questions: in your tests, how many times does it asks for the CAPTCHA (no problem if it's just a guess)? How long does it take to run it (the full dataset)? Cheers